DOC-6614 Pub/sub use case (#3311)
redis/docs: content/develop/use-cases/pub-sub/redis-py/_index.md

* Each `Subscription` gets its own `PubSub` (and therefore its own Redis connection). Sharing one connection across subscribers would couple their lifetimes — closing one would close the channel for the others.
* `run_in_thread` runs a short poll loop (`sleep_time=0.01`) rather than blocking on `listen()`. That's the cleanest way to support `subscription.close()` without leaving a thread stuck inside a blocking socket read.

## Pattern subscriptions with PSUBSCRIBE

`psubscribe()` works the same way but routes messages through `PSUBSCRIBE` so each binding is a glob, not a literal channel name:

```python
hub.psubscribe(name="all-notifications", patterns=["notifications:*"])
hub.psubscribe(name="cache-invalidator", patterns=["cache:invalidate:*"])
```

When a published channel matches a pattern, the dispatch callback receives both the matched channel and the original pattern:

```python
def _dispatch(self, raw: dict) -> None:
    channel = raw.get("channel")   # the actual channel the message was sent to
    pattern = raw.get("pattern")   # the pattern that matched (None for exact subs)
    data    = raw.get("data")
    ...
```

> That distinction is useful for routing: a pattern subscriber can do one thing for the whole hierarchy (e.g., increment a counter) and dispatch on the specific channel within its callback (e.g., "invalidate this region's cache").

Reviewer: https://developers.google.com/style/word-list#eg

```suggestion
That distinction is useful for routing: a pattern subscriber can do one thing for the whole hierarchy (such as increment a counter) and dispatch on the specific channel within its callback (such as "invalidate this region's cache").
```
